### PR TITLE
Fixed order entry; added MO counter chart; removed MOEMA plot

### PR DESCRIPTION
**Pull request by westshawn**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -435,14 +444,18 @@ def OnDataMO(self, slice):
...
+
             # Only check for entry signals during set trade time
             if not self.trade_time:
                 return
 
             # Check for entry signals
             check_for_entries = False
             if self.mo_dir_count is not None:
-                if self.mo_dir_count >= SLOPE_LENGTH:
+                if self.mo_dir_count <= SLOPE_LENGTH:
```
Comment (westshawn) [inline main.py]: Is this change accurate?
Don't we want to look for entries when mo_dir_count > slope?

File: main.py [hunk 2]
```diff
@@ -319,27 +332,28 @@ def OnDataAlgoTotal(self, slice):
             for tick in quote_ticks:
                 # Check for changes and update bid, ask counters
                 if tick.BidPrice != self.last_bid:
```
Comment (westshawn) [inline main.py]: I'm not sure what the intention is here in getting the last_bid and last_ask.

Do we only care about the bid/ask of the last item in the list?
if so, then let's not loop through the list, but just set last_bid and last_ask to the last item in the list directly.

File: main.py [hunk 3]
```diff
@@ -319,27 +332,28 @@ def OnDataAlgoTotal(self, slice):
...
+            #    tick.BidPrice,self.last_bid,tick.AskPrice,self.last_ask, tick.BidSize, tick.AskSize,tick.Quantity))
+
+            #    Count the number of bids and ask.
+                if tick.Value != 0:
+                    self.last_traded_price = tick.Value
+
+            #   self.Debug('last traded price : {}'.format(self.last_traded_price))
+                self.tot_counter += 1
+                self.tot_volume += tick.Quantity
+                if self.last_bid > 0:
```
Comment (westshawn) [inline main.py]: Here we loop through trade_ticks, but don't update last_bid and last_ask in the loop. 
So, bid_counter and ask_counter will get incremented based on the static values of last_bid and last_ask

File: main.py (hunk 1 above)
Comment (jbem777) [reply main.py]: This logic was causing entries to be taken when the slope was over extended. We want entries when the slope is just beginning to turn and for the previous slope direction to have been at least 2 or more bars.

File: main.py (hunk 2 above)
Comment (jbem777) [reply main.py]: 100% concur Shawn. Using the last tick to assign volume needs to be adjusted. I think that's why we are seeing the current issues.

File: main.py (hunk 1 above)
Comment (westshawn) [reply main.py]: This is covered in https://github.com/jumpingvioletshark/BuySellAlgo/issues/12

File: main.py (hunk 2 above)
Comment (westshawn) [reply main.py]: Will be addressed in https://github.com/jumpingvioletshark/BuySellAlgo/issues/15

File: main.py (hunk 3 above)
Comment (westshawn) [reply main.py]: Will be addressed in https://github.com/jumpingvioletshark/BuySellAlgo/issues/16

**Pull request merged**